pontoon/base/formats/silme.py

```diff
@@ -0,0 +1,161 @@
+
+    @fuzzy.setter
+    def fuzzy(self, fuzzy):
+        pass  # We don't use fuzzy in silme
+
+    @property
+    def source(self):
+        return []
+
+    def __eq__(self, other):
+        return self.key == other.key and self.strings.get(None) == other.strings.get(None)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+class SilmeResource(ParsedResource):
+    def __init__(self, parser, path, source_resource=None):
+        self.parser = parser
+        self.path = path
+        self.source_resource = source_resource
+
+        # Preserve entity order via an OrderedDict.
+        if source_resource:
+            self.entities = source_resource.entities
+        else:
+            self.entities = OrderedDict()
+
+        with codecs.open(path, 'r', 'utf-8') as f:
+            self.structure = parser.get_structure(f.read())
+
+        comments = []
+        current_order = 0
+        for obj in self.structure:
+            if isinstance(obj, silme.core.entity.Entity):
+                entity = SilmeEntity(obj, comments, current_order)
+                self.entities[entity.key] = entity
+                current_order += 1
+                comments = []
+            elif isinstance(obj, silme.core.structure.Comment):
+                for comment in obj:
+                    # Silme groups comments together, so we strip
+                    # whitespace and split them up.
+                    lines = unicode(comment).strip().split('\n')
+                    comments += [line.strip() for line in lines]
+
+    @property
+    def translations(self):
+        return self.entities.values()
+
+    def save(self, locale):
+        """
+        Load the source resource, modify it with changes made to this
+        Resource instance, and save it over the locale-specific
+        resource.
+        """
+        if self.source_resource is None:
+            raise SyncError('Cannot save silme resource {0}: No source resource given.'
+                            .format(self.path))
+
+        with codecs.open(self.source_resource.path, 'r', 'utf-8') as f:
+            new_structure = self.parser.get_structure(f.read())
+
+        # Update translations in the copied resource.
+        entities = [
+            SilmeEntity(obj) for obj in new_structure if isinstance(obj, silme.core.entity.Entity)
+        ]
+        for silme_entity in entities:
+            key = silme_entity.key
+
+            translated_entity = self.entities.get(key)
+
+            have_translation = (
+                translated_entity
+                and None in translated_entity.strings
+                and translated_entity != silme_entity
+            )
+            if have_translation:
+                new_structure.modify_entity(key, translated_entity.strings[None])
+            else:
+                # Remove untranslated entity and following newline
+                pos = new_structure.entity_pos(key)
+                new_structure.remove_entity(key)
```

Comment:
This line should be the only thing required, the rest should be moved to Silme. One day.